modules/core/src/main/java/org/apache/synapse/util/MessageHelper.java

```diff
@@ -414,7 +426,9 @@ public static MessageContext cloneMessageContextForAggregateMediator(MessageCont
         }
         // XXX: always this section must come after the above step. ie. after applying Envelope.
         // That is to get the existing headers into the new envelope.
-        JsonUtil.cloneJsonPayload(mc, newMC);
+        if (isCloneJson) {
+            JsonUtil.cloneJsonPayload(mc, newMC);
+        }
```

Comment:
If isCloneJson=false we clone the context, but just place the reference to the original json playload without cloning? What if somebody used payload factory inside iterate mediator? Then on next iteration it is applied on the replaced payload?